src/coreclr/interpreter/naming.cpp

```diff
@@ -0,0 +1,283 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "interpreter.h"
+
+void AppendType(COMP_HANDLE comp, TArray<char>* printer, CORINFO_CLASS_HANDLE clsHnd, bool includeInstantiation);
+void AppendCorInfoType(TArray<char>* printer, CorInfoType corInfoType);
+void AppendTypeOrJitAlias(COMP_HANDLE comp, TArray<char>* printer, CORINFO_CLASS_HANDLE clsHnd, bool includeInstantiation);
+
+void AppendString(TArray<char>& array, const char* str)
+{
+    if (str != nullptr)
+    {
+        size_t strLen = strlen(str);
+        array.Append(str, static_cast<int32_t>(strLen));
+    }
+}
+
+void AppendCorInfoTypeWithModModifiers(TArray<char>* printer, CorInfoTypeWithMod corInfoTypeWithMod)
+{
+    if ((corInfoTypeWithMod & CORINFO_TYPE_MOD_PINNED) == CORINFO_TYPE_MOD_PINNED)
+    {
+        printer->Append("PINNED__", 7);
+    }
+    if ((corInfoTypeWithMod & CORINFO_TYPE_MOD_COPY_WITH_HELPER) == CORINFO_TYPE_MOD_COPY_WITH_HELPER)
+    {
+        printer->Append("COPY_WITH_HELPER__", 17);
+    }
+}
+
+void AppendCorInfoType(TArray<char>* printer, CorInfoType corInfoType)
+{
+    static const char* preciseVarTypeMap[CORINFO_TYPE_COUNT] = {
+        // see the definition of enum CorInfoType in file inc/corinfo.h
+        "<UNDEF>",
+        "void",
+        "bool",
+        "char",
+        "sbyte",
+        "byte",
+        "short",
+        "ushort",
+        "int",
+        "uint",
+        "long",
+        "ulong",
+        "nint",
+        "nuint",
```

Comment:
```suggestion
#ifdef TARGET_64BIT
        "long",
        "ulong",
#else
        "int",
        "uint",
#endif
```

Similarly, to match JIT exactly.